PARTITION_PINNING.md

```diff
@@ -0,0 +1,139 @@
+
+## How It Works
+
+1. **Client Identity**: Each client has a unique numerical identity (client name from Kubernetes StatefulSet)
+2. **Message-based Process Starting**: Instead of creating process instances directly, the client publishes messages with correlation keys that route to specific partitions
+3. **Partition-specific Job Types**: Job workers only subscribe to job types that include their client name, ensuring they only handle jobs from their assigned partitions
+
+## Configuration
+
+### Required Properties
+
+```properties
+# Enable partition pinning
+benchmark.client.enable-partition-pinning=true
+
+# Client name (set automatically from Kubernetes pod metadata)
+benchmark.client.client-name=${POD_NAME}
+
+# Total number of partitions in your Zeebe cluster
+benchmark.client.partitionCount=9
+
+# Total number of client replicas (StatefulSet replicas)
+benchmark.client.replicas=3
+```
+
+### BPMN Process Requirements
+
+When using partition pinning, your BPMN process must have a message start event:
+
+```xml
+<bpmn:startEvent id="StartEvent_MessageStart" name="Start via Message">
+  <bpmn:messageEventDefinition messageRef="Message_StartBenchmarkProcess" />
+</bpmn:startEvent>
+```
+
+Use the provided `benchmark_partition_pinning.bpmn` process or add a message start event to your existing process.
+
+### Kubernetes Configuration
+
+Deploy as a StatefulSet to ensure stable pod identities:
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: camunda-8-benchmark-partitioned
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+        - name: camunda-8-benchmark
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: JAVA_OPTIONS
+              value: >-
+                -Dbenchmark.client.enable-partition-pinning=true
+                -Dbenchmark.client.partitionCount=9
+                -Dbenchmark.client.replicas=3
+                -Dbenchmark.bpmnProcessId=benchmark_partition_pinning
+                -Dbenchmark.bpmnResource=classpath:bpmn/benchmark_partition_pinning.bpmn
+            - name: BENCHMARK_CLIENT_CLIENT_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+```
+
+## Partition Assignment Strategy
+
+The partition assignment follows this logic:
+
+- **More partitions than replicas**: Each client gets assigned to different partitions evenly distributed
+- **More replicas than partitions**: Multiple clients may share the same partition
+- **Equal partitions and replicas**: One-to-one mapping
+
+Examples:
+- 9 partitions, 3 replicas: Client 0 → Partition 0, Client 1 → Partition 3, Client 2 → Partition 6
```

Comment:
This example is still weird. Please list all partitions that each client is working on. Make sure to follow the round robin principle, e.g. Client 0 -> Partitions 0, 3, and 6